src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -79,6 +79,10 @@ public LanguageExpression ConvertExpression(SyntaxBase expression)
                         function.Arguments.Select(a => ConvertExpression(a.Expression)).ToArray());
 
                 case InstanceFunctionCallSyntax instanceFunctionCall:
+                    var namespaceSymbol = context.SemanticModel.GetSymbolInfo(instanceFunctionCall.BaseExpression);
+                    
+                    Assert(namespaceSymbol is NamespaceSymbol, $"BaseExpression must be a NamespaceSymbol, instead got: '{namespaceSymbol?.Kind}'");
```

Comment:
FYI `Debug.Assert` is useful for this kind of scenario (see [here](https://github.com/Azure/bicep/blob/6d734425689be4c9b6b765cf7de789bc0eb9c0cd/src/Bicep.Core/Text/SpellChecker.cs#L70)for an example). No need to change now though